### PR TITLE
arrays-arrays_with_methods-hard-q13

### PR DESCRIPTION
**Pull request by SamamthaH**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [x] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
Update test cases to use arrayUtil

**Comment by SamamthaH** [comment]
> Update test cases to use arrayUtil

This question also doesn't take arrays for inputs.

File: src/arrays/arrays_with_methods/hard/q13/ArrayngeOfPrimes.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+public class ArrayngeOfPrimes {
+    public static void main(String[] args) {
+        Scanner input = new Scanner(System.in);
+        System.out.println("Enter the size of the array: ");
+        int n = input.nextInt();
+        int[] arr = new int[n];
+        int primeCount = 0;
+        int elem = 2;
+        while (primeCount < n) {
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q13/ArrayngeOfPrimes.java:29]: For this one, we can't make sure they use an array, so we should change this question.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/arrays_with_methods/hard/q13/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        return Stream.of(
+                ArrayUtil.generateAscendingArray(1, n),
+                ArrayUtil.generateRandomArray(1, 10000, n),
+                ArrayUtil.generateRandomArray(1, 10, n)
+        );
+    }
+
+    static Stream<int[]> getPrimesInputProvider() {
+        return Stream.of(
+                ArrayUtil.generateAscendingArray(1, 100),
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q13/MainTest.java:50]: You should add some additional test cases for some larger prime numbers

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/arrays_with_methods/hard/q13/ArrayngeOfPrimes.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+/*
+Write a program that asks the user to enter an array of 10 integers. Then, the program passes the array to a method
+called getPrimes which identifies all the primes in the given array as an integer n and returns them in a new array of
+size n. The main method should then print the array of primes. Note that prime numbers start from 2 since they must be
+divisible only by 1 and themselves. Your getPrimes method should be generalized to work with an array of any size.
+
+Sample output:
+    Enter an array of 10 integers:
+    1 2 3 4 5 6 7 8 9 10
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q13/ArrayngeOfPrimes.java:11]: I feel like we should also add context to the sample output

Comment (Hedgemon4) [reply src/arrays/arrays_with_methods/hard/q13/ArrayngeOfPrimes.java:11]: And print the numbers from the prime array on the same line

File: src/arrays/arrays_with_methods/hard/q13/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                ArrayUtil.generateRandomArray(1, 10000, 35),
+                ArrayUtil.generateRandomArray(1, 10, 10000),
+                ArrayUtil.generateRandomArray(2, 5, 1),
+                ArrayUtil.generateRandomArray(2, 5, 2),
+                ArrayUtil.generateAscendingArray(1, 250, 2),
+                new int[]{5, 7, 5, 2},
+                new int[]{1, 4, 6, 8, 9, 10, 12, 14, 21, 66, 12, 4},
+                new int[]{15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1},
+                new int[]{3169, 1999, 4793, 5783, 7919, 6311, 672832, 733},
+                new int[]{5, 53, 157, 173, 211, 257, 263, 3731367, 1511, 1747, 1753, 1907, 2287, 2417, 3637, 3733, 4013, 4409, 4457, 4597, 4657, 4691, 4993, 5107, 5113, 5393},
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q13/MainTest.java:60]: Are these all primes? If so they should not all be, you should throw in some numbers that are one off of being a prime.

File: src/arrays/arrays_with_methods/hard/q13/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    @ParameterizedTest
+    @MethodSource("getPrimesInputProvider")
+    void correctGetPrimesMethod(int[] in) throws Throwable {
+        int[] expected = solution(in);
+        Object[][] arguments = {
+                {in, int[].class}
+        };
+        MethodTest m = new MethodTest(ArrayngeOfPrimes.class, "getPrimes", arguments);
+        Object output = m.callMethod();
+        CustomAssertions._assertArrayEquals(expected, output, "Your getPrimes method does not correctly collect and return all prime numbers.");
```
Comment (Hedgemon4) [inline src/arrays/arrays_with_methods/hard/q13/MainTest.java:75]: I think that "collect" does not give enough information. You should phrase it more in the context of returning an array of all the prime numbers

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/arrays_with_methods/hard/q13/MainTest.java (hunk 4 above)
Comment (SamamthaH) [reply src/arrays/arrays_with_methods/hard/q13/MainTest.java:60]: 3731367 is not a prime, I'll throw in one number that is also slightly off

**Review by Hedgemon4 — APPROVED** [review]

**Pull request merged**